nav2_controller/include/nav2_controller/plugins/stopped_goal_checker.hpp

```diff
@@ -55,7 +55,7 @@ class StoppedGoalChecker : public SimpleGoalChecker
   StoppedGoalChecker();
   // Standard GoalChecker Interface
   void initialize(
-    const rclcpp_lifecycle::LifecycleNode::SharedPtr & nh,
+    const rclcpp_lifecycle::LifecycleNode::WeakPtr & parent,
```

Comment:
These are ABI breaking changes that cannot be backported

Reply:
Okay. The reason I included PR #1900 is because without it I was getting a "marked ‘override’, but does not override" error when building. Because in smac_planner_2d, the configure() function uses WeakPtr. But in the base class GlobalPlanner, configure() uses SharedPtr on the foxy-devel branch. 

Any way around this?

Reply:
I think you need to do the following:
- Go through this PR and revert anything that is weak ptr related (API, the `logger_`, and `clock_` additions)
- Modify SMAC to use the shared ptr vs the weak ptr since that's what Foxy uses

I think the rest should be good, might have a couple of small things to deal with but it should be minimal

Reply:
Thank you. I committed those changes.